src/main/java/de/dennisguse/opentracks/ui/markers/MarkerListAdapter.java

```diff
@@ -0,0 +1,306 @@
+            this.marker = marker;
+
+            photo.setVisibility(marker.hasPhoto() ? View.VISIBLE : View.GONE);
+            textGradient.setVisibility(marker.hasPhoto() ? View.VISIBLE : View.GONE);
+            if (marker.hasPhoto()) {
+                int height = getPhotoHeight();
+                ViewGroup.LayoutParams params = photo.getLayoutParams();
+                params.height = height;
+                photo.setLayoutParams(params);
+                textGradient.setLayoutParams(params);
+
+                asyncLoadPhoto(photo, marker.getPhotoURI(), marker.getId().id());
+            }
+
+            name.setText(marker.getName());
+
+            {
+                OffsetDateTime offsetDateTime = OffsetDateTime.ofInstant(marker.getTime(), timeZone);
+                String dateContent = StringUtils.formatDateTodayRelative(context, offsetDateTime);
+                String pattern = "HH:mm";
+                if (!offsetDateTime.getOffset().equals(OffsetDateTime.now().getOffset())) {
+                    pattern = "HH:mm x";
+                }
+                String timeContent = offsetDateTime.format(DateTimeFormatter.ofPattern(pattern));
+
+                date.setText(dateContent);
+                time.setText(timeContent);
+            }
+
+            String categoryDescription = StringUtils.getCategoryDescription(marker.getCategory(), marker.getDescription());
+            timeDistance.setText(categoryDescription);
+
+            setSelected(selection.get(getLayoutPosition()));
+        }
+
+        public void setSelected(boolean isSelected) {
+            selection.put(getLayoutPosition(), isSelected);
+            view.setActivated(isSelected);
+        }
+
+        public long getId() {
+            return marker.getId().id();
+        }
+
+        @Override
+        public void onClick(View view) {
+            if (selectionMode) {
+                setSelected(!view.isActivated());
+                actionMode.invalidate();
+            } else {
+                Intent intent = IntentUtils.newIntent(context, MarkerDetailActivity.class)
+                        .putExtra(MarkerDetailActivity.EXTRA_MARKER_ID, marker.getId());
+                context.startActivity(intent);
+            }
+        }
+
+        @Override
+        public boolean onLongClick(View view) {
+            actionMode = context.startSupportActionMode(MarkerListAdapter.this);
+            setSelected(true);
+            return true;
+        }
+    }
+
+    /**
+     * It loads the photoUrl in the imageView from view.
+     * It takes the photo from cache or from storage if isn't in the cache.
+     *
+     * @param imageView view object where photo will be loaded.
+     * @param photoUri  photo's url.
+     * @param id        marker's id where photo belong.
+     */
+    private void asyncLoadPhoto(ImageView imageView, Uri photoUri, long id) {
+
+        Bitmap bm = memoryCache.get(id);
+        imageView.setImageBitmap(bm);
```

Comment:
Set only if not null.

Reply:
I wouldn't do that. The View Holder could still be containing an image from a previous list item, that is not visible anymore. When this new Marker doesn't have an image, the old one would still remain